src/main/kotlin/org/taktik/icure/asyncdao/impl/PatientDAOImpl.kt

```diff
@@ -279,9 +279,10 @@ class PatientDAOImpl(
 			endKey = ComplexKey.of(healthcarePartyId, "\ufff0")
 		}
 
-		val viewQuery = createQuery(client, "by_hcparty_address").startKey(startKey).endKey(endKey).includeDocs(false)
-
-		emitAll(client.queryView<Array<String>, String>(viewQuery).mapNotNull { it.value })
+		val viewQuery = createQuery(client, "by_hcparty_address_postalcode_housenumber").startKey(startKey).endKey(endKey).includeDocs(false)
+		emitAll(client.queryView<Array<String>, String>(viewQuery).filter {
```

Comment:
No: In your view, you're putting the postal code & the houseNumber in the arguments of your key. 
If you want to filter your results efficiently, take advantage of CouchDB and put your values into the startKey & endKey instead of filtering with Kotlin.

Reply:
I'll still have to filter with kotlink because Couchdb will return too many results, but i'll add them to the start and endkey